src/RequestProcessor.js

```diff
@@ -85,6 +70,32 @@ class RequestProcessor {
       guard = false;
     });
   }
+
+  processEvents(fromBlock, toBlock) {
+    logger.info(`Start processing events from blocks ${fromBlock}-${toBlock}`);
+
+    this.oracleContract.getPastEvents(
+      'DataRequested',
+      { fromBlock, toBlock },
+    ).then(events => events.forEach((event) => {
+      logger.info(`DataRequested event: ${JSON.stringify(event)}`);
+
+      const request = Object.assign({}, event.returnValues, { startedAt: Date.now() });
+
+      this.requestDao.saveRequest(request);
+    }));
+
+    this.oracleContract.getPastEvents(
+      'DelayedDataRequested',
+      { fromBlock, toBlock },
+    ).then(events => events.forEach((event) => {
+      logger.info(`DelayedDataRequested event: ${JSON.stringify(event)}`);
+
+      const request = Object.assign({}, event.returnValues, { startedAt: Date.now() });
+      request.validFrom *= 1000;
```

Comment:
You could add a comment explaining that Solidity operates on epoch seconds and JS operates on epoch millis.